Prepare templates

izdeby opened the pull request:
This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30983 [WIP] Updated codegen to remove TensorOptions from generated code
* **#30982 Prepare templates**
* #30981 Introduce BackendSelect dispatch key

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

izdeby committed:
Prepare templates

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------



[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit e3bce29:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e3bce298cbccde7cafa43af44cbde3029d869699).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3935002) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** "pytorch android gradle build" ([details](https://dr.pytorch.org/build-details.html?build_id=73657186))
```
sys	0m0.148s 
++ docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-e3bce298cbccde7cafa43af44cbde3029d869699-android-x86_32 
+ export id_x86_32=b1a8859c529b389f9c0de69b8f5921ef1ee17725d87ac3fd2115f6aa9c5592b4 
+ id_x86_32=b1a8859c529b389f9c0de69b8f5921ef1ee17725d87ac3fd2115f6aa9c5592b4 
+ export 'COMMAND=((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1' 
+ COMMAND='((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1' 
+ echo '((echo' '"source' './workspace/env"' '&&' echo '"sudo' chown -R jenkins 'workspace")' '|' docker exec -u jenkins -i '"$id_x86_32"' 'bash)' '2>&1' 
+ ts 
+ unbuffer bash ./command.sh 
+ docker pull 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-e3bce298cbccde7cafa43af44cbde3029d869699-android-arm-v7a 
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-e3bce298cbccde7cafa43af44cbde3029d869699-android-arm-v7a not found 
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]

izdeby committed:
Update on "Prepare templates"

This stack is a first step toward an effort to fix, clean up and simplify code generation logic.  Please see the master [task](https://github.com/pytorch/pytorch/issues/30405) to see related discussions and all the known issues.

Main focus of these changes is TensorOptions in code generation. 
Goals:
- Remove TensorOptions from generated code wherever it's possible. Leave it only in python/C++ API layers.
- Refactor TensorOptions logic to a single place.
- Log all discovered issues.

Non goals:
- Fix Everything! 
- Remove all the hacks in code generation scripts.
- Clean up and defector all code generation scripts.

-----------
In this PR:
Updating the templates.

-----------

Differential Revision: [D18912680](https://our.internmc.facebook.com/intern/diff/D18912680)

[ghstack-poisoned]
